metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -104,115 +130,277 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+				"resource_pool": mapstr.M{
+					"id": virtualMachine.ResourcePool.Value,
+				},
+				"cpu": mapstr.M{
+					"reserved": virtualMachine.Summary.Config.CpuReservation,
+					"cores":    virtualMachine.Summary.Config.NumCpu,
+				},
+				"disks": mapstr.M{
+					"count": virtualMachine.Summary.Config.NumVirtualDisks,
+				},
+				"storage": mapstr.M{
+					"committed":  virtualMachine.Summary.Storage.Committed,
+					"uncommited": virtualMachine.Summary.Storage.Uncommitted,
+					"total":      virtualMachine.Summary.Storage.Committed + virtualMachine.Summary.Storage.Uncommitted,
+				},
+				"heartbeat_status": virtualMachine.GuestHeartbeatStatus,
+				"connection_state": virtualMachine.Summary.Runtime.ConnectionState,
+				"memory": mapstr.M{
+					"overhead":   virtualMachine.Summary.Runtime.MemoryOverhead,
+					"total_size": virtualMachine.Summary.Config.MemorySizeMB,
+					"reserved":   virtualMachine.Summary.Config.MemoryReservation,
+				},
+				"power_state":                   virtualMachine.Summary.Runtime.PowerState,
+				"snapshot_consolidation_needed": virtualMachine.Summary.Runtime.ConsolidationNeeded,
+				"vmx_path":                      virtualMachine.Summary.Config.VmPathName,
 			}
-			event.Put("memory.total.guest.bytes", totalMemory)
-			event.Put("memory.free.guest.bytes", freeMemory)
-		}
 
-		if host := vm.Summary.Runtime.Host; host != nil {
-			event["host.id"] = host.Value
-			hostSystem, err := getHostSystem(ctx, c, host.Reference())
-			if err == nil {
-				event["host.hostname"] = hostSystem.Summary.Config.Name
+			// Add resource pool name and path if available, if not, set the rpRefreshRequired var to true
+			resourcePool, ok := m.RpMap[virtualMachine.ResourcePool.Value]
+			if !ok {
+				rpRefreshRequired = true
+				m.Logger().Debugf("resource pool with id %s not found, will refresh pool information at end of current run", virtualMachine.ResourcePool.Value)
 			} else {
-				m.Logger().Debug(err.Error())
+				event.Put("resource_pool.name", resourcePool.Name)
+				event.Put("resource_pool.path", resourcePool.InventoryPath)
 			}
-		} else {
-			m.Logger().Debug("'Host', 'Runtime' or 'Summary' data not found. This is either a parsing error " +
-				"from vsphere library, an error trying to reach host/guest or incomplete information returned " +
-				"from host/guest")
-		}
 
-		// Get custom fields (attributes) values if get_custom_fields is true.
-		if m.GetCustomFields && vm.Summary.CustomValue != nil {
-			customFields := getCustomFields(vm.Summary.CustomValue, customFieldsMap)
-
-			if len(customFields) > 0 {
-				event["custom_fields"] = customFields
+			if virtualMachine.Datastore != nil {
+				var datastores []DatastoreInfo
+				for _, ds := range virtualMachine.Datastore {
+					datastore, ok := m.DsMap[ds.Value]
+					if !ok {
+						dsRefreshRequired = true
+						m.Logger().Debugf("datastore with id %s not found, will refresh datastore information at the end of the run")
```

Comment:
```suggestion
						m.Logger().Debugf("datastore with id %s not found, will refresh datastore information at the end of the run")
```

used %s but no arg passed to Debugf?

Reply:
not sure how I missed that, datastore value added to the debug statement